src/plugins/intel_gpu/tests/test_cases/concatenation_gpu_test.cpp

```diff
@@ -1285,4 +1285,154 @@ INSTANTIATE_TEST_SUITE_P(smoke,
+
+        auto weights_lay = cldnn::layout(data_type, cldnn::format::bfyx, tensor(batch(output_f), feature(output_f)));
+        auto weights_mem = engine.allocate_memory(weights_lay);
+        auto& stream = get_test_stream();
+        weights_mem->fill(stream);
+        stream.finish();
+        {
+            cldnn::mem_lock<Type> weights_ptr(weights_mem, stream);
+            for (size_t fi = 0; fi < output_f; ++fi) {
+                auto coords = tensor(batch(fi), feature(fi), spatial(0, 0, 0, 0));
+                auto offset = weights_lay.get_linear_offset(coords);
+                weights_ptr[offset] = static_cast<Type>(1.f);
+            }
+        }
+
+        topology.add(data("weights" , weights_mem));
+        topology.add(convolution("conv", input_info("eltwise"), { "weights" }));
+        topology.add(concatenation("concat", {input_info("conv"), input_info("eltwise")}, 1));
+        topology.add(reorder("reorder", input_info("concat"), layout(data_type, format::bfyx, {(int32_t)batch_num, (int32_t)(output_f * 2), (int32_t)input_y, (int32_t)input_x})));
+
+        network concat_network(engine, topology, config);
+        for (size_t i = 0; i < 2; i++) {
+            concat_network.set_input_data(input_ids[i], in_memory[i]);
+        }
+        concat_network.execute();
+
+        bool concat_opt_enabled = config.get_property(ov::intel_gpu::optimize_data);
+        bool concat_opt_result = std::static_pointer_cast<concatenation_inst>(concat_network.get_primitive("concat"))->node->can_be_optimized();
+
+        // If sibling is using onednn impl and batch > 1, the onednn impl cannot process the implicit concat'ed buffer.
+        // Onednn impls can process implicit concat'ed buffer only through buffer pointer manipulation.
+        if (concat_opt_enabled && batch_num > 1) concat_opt_result = !concat_opt_result;
+        EXPECT_EQ(concat_opt_enabled, concat_opt_result);
+
+        return concat_network.get_output("reorder").get_memory();
+    }
+
+    std::vector<std::vector<std::vector<std::vector<std::vector<Type>>>>> generate_input() {
+        const size_t batch_num = testing::get<0>(GetParam());
+        const std::vector<size_t> in_features = testing::get<1>(GetParam());
+        const size_t input_y = testing::get<2>(GetParam());
+        const size_t input_x = testing::get<3>(GetParam());
+
+        std::vector<std::vector<std::vector<std::vector<std::vector<Type>>>>> inputs(2);
+        for (size_t i = 0; i < 2; ++i) {
+            inputs[i] = generate_random_4d<Type>(batch_num, in_features[0], input_y, input_x, -1, 1);
+        }
+        return inputs;
+    }
+
+    void test(format::type fmt) {
+        auto& engine = get_test_engine();
+        auto& stream = get_test_stream();
+        if (!engine.get_device_info().supports_immad) {
+            // This case is only for device that uses onednn.
+            return;
+        }
+        auto input = generate_input();
+
+        // implicit concat
```

Comment:
this is a misleading comment. It is implicit only when b=1, right?

Reply:
Fixed